editor/src/components/navigator/navigator-item/component-picker-context-menu.tsx

```diff
@@ -441,6 +446,34 @@ function insertComponentPickerItem(
       ]
     }
 
+    if (isReplaceTarget(insertionTarget)) {
+      if (
+        MetadataUtils.isJSXMapExpression(EP.parentPath(target), metadata) &&
+        elementWithoutUID.type === 'JSX_MAP_EXPRESSION'
+      ) {
+        return [
```

Comment:
I think we should show a toast for `elementWithoutUID.type !== 'JSX_ELEMENT'` because now Fragment and Conditional fail silently, and only List shows a message.